### ISSUE TITLE
"Unable to verify integrity of data upload" when uploading data to S3 with a trailing newline

### ISSUE THREAD

**Issue by rotoclone** [localstack/localstack#1305]
I am using the Java AWS client to upload data to a localstack S3 bucket in a JUnit test that uses `LocalstackDockerTestRunner`.

When doing a PutObject request with data that has a trailing newline, I'm seeing the following error:
```
com.amazonaws.SdkClientException: Unable to verify integrity of data upload. Client calculated content hash (contentMD5: iMAcxsPQ9H8OR5NhVfjP5g== in base 64) didn't match hash (etag: 1a368e75679a614ed41f60f6a1be0cef in hex) calculated by Amazon S3.  You may need to delete the data stored in Amazon S3. (metadata.contentMD5: null, md5DigestStream: com.amazonaws.services.s3.internal.MD5DigestCalculatingInputStream@57c03d88, bucketName: test-bucket, key: new/9fca0482-fea8-40f4-adcc-fe21464e2500/file.txt)
	at com.amazonaws.services.s3.AmazonS3Client.uploadObject(AmazonS3Client.java:1812)
	at com.amazonaws.services.s3.AmazonS3Client.putObject(AmazonS3Client.java:1746)
	at com.amazonaws.services.s3.AmazonS3Client.putObject(AmazonS3Client.java:1687)
```

I'm using the latest docker image and localstack-utils 0.1.21. This just started happening today after the update to the docker image that fixed https://github.com/localstack/localstack/issues/1293

Here's a minimal project that demonstrates the error: https://github.com/rotoclone/localstack-error

**Comment by przemek-lichota** [comment]
Hi, I'am also facing with this error while issuing AmazonS3Client.putObject request. But uploading file using TransferManager.upload works fine.

**Comment by MikkCZ** [comment]
Hi.

Did you have any success working around this issue? We are hitting issues with LocalStack S3 for the last two days and confirmed today it happens for files ending with `0A` only, so likely the same issue you hit.

We started with image from DockerHub in version 0.9.2 and later 0.9.6, but always the same problem. For some reason 0.7.0 seem to work, however we are not certain we want to use such old version.

**Comment by rotoclone** [comment]
@MikkCZ My workaround is to not upload stuff with a trailing newline, but that of course only works if you don't rely on the trailing newline being there.

**Comment by przemek-lichota** [comment]
Try to set chunked encoding disabled to true while instantiating AmazonS3 client. It works for me in version 0.9.5:
```
AmazonS3ClientBuilder.standard()
				.withCredentials(...)
				.withChunkedEncodingDisabled(true)
				.withPathStyleAccessEnabled(true)
				.withEndpointConfiguration(...)
				.build();
```

**Comment by MikkCZ** [comment]
Thank you both. In our case we cannot prevent `0A` from being at the end of file. It's binary data in Avro format and stripping it (which happens in LocalStack when the file is being saved) invalidated the whole file.

We will try to disable chunked encoding a let you know if it solved the issue.

**Comment by MikkCZ** [comment]
Thanks @przemek-lichota , disabling chunked encoding helped. Now it works with 0.10.0 too.

**Comment by whummer** [comment]
Looks like the issue with chunked encoding should be fixed in the latest version, as confirmed in #1571 . Please re-open this issue if this problem persists. Thanks
